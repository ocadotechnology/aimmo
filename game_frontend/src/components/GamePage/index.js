import React, { Component } from 'react'
import styled from 'styled-components'
import IDE from 'components/IDE'
import Game from 'components/Game'

export const GamePageLayout = styled.div`
  display: grid
  grid-template: 80px 1fr 150px / 1fr 1fr
  grid-template-areas: "ide-menu game-menu"
                       "ide-editor game-view"
                       "ide-console game-view";
  width: 100vw
  height: 100vh
`

export default class GamePage extends Component {
  render () {
    return (
      <GamePageLayout>
        <IDE />
        <Game />
      </GamePageLayout>
    )
  }
}
