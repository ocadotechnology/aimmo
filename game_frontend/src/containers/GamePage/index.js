import React, { Component, Fragment } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { actions } from 'features/GhibliMovies'
import Movie from 'components/Movie'
import Button from 'components/Button'

export class GamePage extends Component {
  render () {
    const movieItems = this.props.movies.map(movie =>
      <Movie key={movie.id} movie={movie} />
    )

    return (
      <Fragment>
        <div>Welcome to the AIMMO game screen</div>
        <Button
          id='fetch-movies-button'
          onClick={() => { this.props.fetchMovies() }}>
          Get the GhibliMovies!
        </Button>
        <ul>
          {movieItems}
        </ul>
      </Fragment>
    )
  }
}

GamePage.propTypes = {
  movies: PropTypes.array,
  fetchMovies: PropTypes.func
}

const mapStateToProps = state => {
  return {
    movies: state.movieReducer.movies
  }
}

const mapDispatchToProps = {
  fetchMovies: actions.fetchMovies
}

export default connect(mapStateToProps, mapDispatchToProps)(GamePage)
