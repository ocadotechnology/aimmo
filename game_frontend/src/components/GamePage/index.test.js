/* eslint-env jest */
import React from 'react'
import GamePage, { GamePageLayout } from 'components/GamePage'
import createShallowWithTheme from 'testHelpers/createShallow'
import { shallow } from 'enzyme'

describe('<GamePage />', () => {
  it('matches snapshot', () => {
    const tree = shallow(<GamePage />)

    expect(tree).toMatchSnapshot()
  })
})

describe('<GamePageLayout />', () => {
  it('matches snapshot', () => {
    const tree = createShallowWithTheme(<GamePageLayout />)

    expect(tree).toMatchSnapshot()
  })
})
