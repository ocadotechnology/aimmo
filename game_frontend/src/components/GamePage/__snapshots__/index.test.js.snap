// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GamePage /> matches snapshot 1`] = `
<styled.div>
  <NavigationBar />
  <Connect(IDE) />
  <Connect(Game) />
</styled.div>
`;

exports[`<GamePageLayout /> matches snapshot 1`] = `
.c0 {
  display: grid;
  grid-template: auto 1fr 250px / 1fr 1fr;
  grid-template-areas: "navigation-bar navigation-bar" "ide-editor game-view" "ide-console game-view";
  width: 100vw;
  height: 100vh;
}

<div
  className="c0"
/>
`;
